docs/codegen.md

```diff
@@ -1,45 +1,30 @@
-@ -1,216 +0,0 @@
 # CodeGen Documentation
 
-CodeGen is the process for auto-generating language-specific, strongly-typed ASTs to be used in [Semantic](https://github.com/github/semantic-code/blob/d9f91a05dc30a61b9ff8c536d75661d417f3c506/design-docs/precise-code-navigation.md).
+CodeGen is the process for auto-generating language-specific, strongly-typed ASTs to be used in [Semantic](https://github.com/github/semantic-code/blob/d9f91a05dc30a61b9ff8c536d75661d417f3c506/design-docs/precise-code-navigation.md). 
+Since it is a critical component of Semantic's language support process, we recommend reading [these docs](https://github.com/github/semantic/blob/master/docs/adding-new-languages.md) first, as they provide an overview of the pipeline CodeGen supports.
```

Comment:
I think we should make all these github links permalinks so that they don't break when we move files around. You can always replace the sha with `master` if you need to, but it's hard to go the other way.